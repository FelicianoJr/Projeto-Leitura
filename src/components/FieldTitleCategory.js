import React from "react";
import { Field } from "redux-form";

const FieldTitleCategory = ({ categories = [] }) => {
  return (
    <div>
      <div className="form-group">
        <label>Titulo</label>
        <Field
          name="title"
          id="title"
          className="form-control"
          component="input"
          type="text"
        />
      </div>
      <div className="form-group">
        <label>Categoria</label>
        <Field
          name="category"
          id="category"
          className="form-control"
          component="select"
        >
          <option />
          {categories.map((category, index) => (
            <option key={index} value={category.name}>
              {" "}
              {category.name}
            </option>
          ))}
        </Field>
      </div>
    </div>
  );
};

export default FieldTitleCategory;
