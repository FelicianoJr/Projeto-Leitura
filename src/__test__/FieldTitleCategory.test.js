import React from "react";
import { shallow } from "enzyme";
import FieldTitleCategory from "../components/FieldTitleCategory";

const setup = () => {
  const component = shallow(<FieldTitleCategory />);

  return {
    component: component,
    formGroup: component.find(".form-group"),
    label: component.find("label"),
    input: component.find("Field")
  };
};

describe("<FieldTitleCategory/>", () => {

  it("should rende  two <FormGroup>", () => {
    const { formGroup } = setup();
    expect(formGroup.length).toEqual(2);
  });

  it("should rende two <Label> ", () => {
    const { label } = setup();
    expect(label.at(0).text()).toEqual("Titulo");
    expect(label.at(1).text()).toEqual("Categoria");
  });

  it("should rende two length input ", () => {
    const { input } = setup();
    expect(input.length).toEqual(2);
  });
});
