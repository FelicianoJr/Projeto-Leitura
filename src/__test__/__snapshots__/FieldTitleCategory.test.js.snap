// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldTitleCategory/> should component Snapshot  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FieldTitleCategory />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="form-group"
>
          <label>
                    Titulo
          </label>
          <Field
                    className="form-control"
                    component="input"
                    id="title"
                    name="title"
                    type="text"
          />
</div>,
        <div
          className="form-group"
>
          <label>
                    Categoria
          </label>
          <Field
                    className="form-control"
                    component="select"
                    id="category"
                    name="category"
          >
                    <option />
          </Field>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Titulo
</label>,
            <Field
              className="form-control"
              component="input"
              id="title"
              name="title"
              type="text"
/>,
          ],
          "className": "form-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Titulo",
            },
            "ref": null,
            "rendered": "Titulo",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "form-control",
              "component": "input",
              "id": "title",
              "name": "title",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Categoria
</label>,
            <Field
              className="form-control"
              component="select"
              id="category"
              name="category"
>
              <option />
</Field>,
          ],
          "className": "form-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Categoria",
            },
            "ref": null,
            "rendered": "Categoria",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <option />,
                Array [],
              ],
              "className": "form-control",
              "component": "select",
              "id": "category",
              "name": "category",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "option",
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group"
>
            <label>
                        Titulo
            </label>
            <Field
                        className="form-control"
                        component="input"
                        id="title"
                        name="title"
                        type="text"
            />
</div>,
          <div
            className="form-group"
>
            <label>
                        Categoria
            </label>
            <Field
                        className="form-control"
                        component="select"
                        id="category"
                        name="category"
            >
                        <option />
            </Field>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Titulo
</label>,
              <Field
                className="form-control"
                component="input"
                id="title"
                name="title"
                type="text"
/>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Titulo",
              },
              "ref": null,
              "rendered": "Titulo",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "form-control",
                "component": "input",
                "id": "title",
                "name": "title",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Categoria
</label>,
              <Field
                className="form-control"
                component="select"
                id="category"
                name="category"
>
                <option />
</Field>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Categoria",
              },
              "ref": null,
              "rendered": "Categoria",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <option />,
                  Array [],
                ],
                "className": "form-control",
                "component": "select",
                "id": "category",
                "name": "category",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "option",
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
